RDKBACCL-1286 : Addition of AP capability messages

AbijithNayak opened the pull request:
Adding the AP capability request and response message to fetch the capabilities from the agent.

AbijithNayak committed:
RDKBACCL-1286 : Addition of AP capability messages

Adding the AP capability request and response message
to fetch the capabilities from the agent.

amarnathhullur commented on src/em/capability/em_capability.cpp:
```diff
@@ -851,39 +940,468 @@ int em_capability_t::handle_bsta_cap_report(unsigned char *buff, unsigned int le
+                break;
+            }
+            if (found_radio == false) {
+                // do not update anything and return error
```
Print appropriate error before returning

AbijithNayak replied:
Added the changes

AbijithNayak committed:
Sending topology and channel pref query

This change is added to send the topology and channel preference query message
on first radio only.

AbijithNayak commented on src/em/capability/em_capability.cpp:
```diff
@@ -851,39 +940,468 @@ int em_capability_t::handle_bsta_cap_report(unsigned char *buff, unsigned int le
+                    break;
+                }
+                if (found_bss == false) {
+                    // do not update anything and return error
```
added the changes

AbijithNayak commented on src/em/capability/em_capability.cpp:
```diff
@@ -45,28 +45,40 @@
-int em_capability_t::create_ap_cap_report_msg(unsigned char *buff)
+int em_capability_t::send_ap_cap_report_msg(unsigned char *dst, unsigned short msg_id)
 {
+    unsigned char buff[MAX_EM_BUFF_SZ + MAX_EM_BUFF_SZ];
```
The message buff was going out of bound as there are many tlvs involved in this message. Hence I have increased it.
